ecllibrary/std/Date.ecl

```diff
@@ -56,6 +56,47 @@ END;
 EXPORT Timestamp_t := INTEGER8;
 
 
+/**
+ * Helper function that ensures that year format parameters in output
+ * format strings (%Y) will emit years that are at least four digits, with
+ * leading zeroes if needed.  Several RFCs mandates a four-digit year to avoid
+ * ambiguities with two-digit years that may be interpreted as being near the
+ * current century.  strftime() is the operating system call we use to format
+ * dates, and they process %Y differently.  It so happens that, for those that
+ * do not pad their year, it is possible to specify a precision within the
+ * format and force the padding to four digits (this is a glibc extension).
+ * That is the purpose of this function:  On those OS versions, insert the
+ * precision format where needed.
+ *
+ * As of this writing, here is how our supported operating systems natively
+ * deal with small years (e.g. 11) and the %Y directive by default:
+ *
+ * macOS:	0011
+ * Windows:	0011
+ * Linux:	11
+ *
+ * All support Linux distributions have glibc, which provides support for
```

Comment:
What happens if the system is compiled using clang?

Reply:
Good question.  I installed Clang 7.0 on my Ubuntu 18.04 system and used (basically) the same code I used on Windows for testing (which is just spitting out the formatted year value from a valid struct tm).  The result was identical to gcc's result:  Using %Y, small years were returned without leading zeroes, and the precision specifier was supported (so %04Y did pad the result).

TL;DR version:  Clang appears to work the same as gcc, which makes sense since it tries to emulate the Gnu toolchain.  Apple may have modified their version of LLVM/Clang so that the leading zeroes are included.